### ISSUE TITLE
Set System Reserved as percentage

### ISSUE THREAD

**Issue by rrmarq** [kubernetes-sigs/karpenter#1595]
### Description

**What problem are you trying to solve?**
Dynamic overhead calculation for O.D resources reservation.

**How important is this feature to you?**
After having some problems with instances running out of capacity until kubelet get killed by the os, letting them with a “zombie node”, we started to reserve resources for the O.S calculating a percentage based on the instance size and setting the value in the userData. 
Setting the value in the userData might make Karpenter calculate the instances resources wrongly ending up and bad instance selections.
An alternative is setting this value in `nodepool.spec.template.spec.kubelet`, however, using this options it is only possible with static values. 
As we have nodepools with different sizes, we would like to set the value overhead as percentage, without having to do it via userData.

**Comment by bnu0** [comment]
We'd love to be able to make these a percentage. We use reserved memory to ensure sufficient burst capacity across our fleet, and achieve higher density by keeping requests accurate. The static nature of these settings means we cannot allocate instances fully dynamically, as the values do not make sense for significantly larger or smaller nodes.

**Comment by rrmarq** [comment]
This is a important request, please, could you give information regarding the team will work on that?

**Comment by rschalo** [comment]
Is there a cloudprovider specific issue that has been also cut for this? We can reopen it in the relevant cloudprivder since this setting is no longer in karpenter core.

**Comment by rrmarq** [comment]
The cloudprovider is AWS.

**Comment by jonathan-innis** [comment]
@rrmarq I think what @rschalo is getting at is that this is now configuration on the NodeClass side -- now, I do think that informing the scheduler of this info is something that we could consider doing in https://github.com/kubernetes-sigs/karpenter/pull/1305 but actually setting the value may be difficult for providers -- specifically thinking about the AWS side, setting different values for systemReserved is going to be really tricky because this basically requires a different kubeletConfiguration setting per instance-type (or at least per instance type size) -- right now, each of these different calculations is going to map to a different userData -- one way we might be able to get around this is by setting a bash function to calculate the reservation numbers ourselves at runtime or somehow pushing those reservation numbers into a map where we look-up the instance type at runtime and set the kubeletconfig accordingly

**Comment by likku123** [comment]
> [@rrmarq](https://github.com/rrmarq) I think what [@rschalo](https://github.com/rschalo) is getting at is that this is now configuration on the NodeClass side -- now, I do think that informing the scheduler of this info is something that we could consider doing in [#1305](https://github.com/kubernetes-sigs/karpenter/pull/1305) but actually setting the value may be difficult for providers -- specifically thinking about the AWS side, setting different values for systemReserved is going to be really tricky because this basically requires a different kubeletConfiguration setting per instance-type (or at least per instance type size) -- right now, each of these different calculations is going to map to a different userData -- one way we might be able to get around this is by setting a bash function to calculate the reservation numbers ourselves at runtime or somehow pushing those reservation numbers into a map where we look-up the instance type at runtime and set the kubeletconfig accordingly

Yes, I think this is a legitimate request if there are multiple instance types available in the nodepool . One should not hardcoding the values because kubelet working condition differs based on the number of pods running in an instance. Values should be different based on the instance type. At least specying in % will help on the guarantee resource based on the instance type spinned

**Comment by Caleb-Bugnacki** [comment]
Any updates on this?
